pkg/middleware/org_redirect.go

```diff
@@ -41,7 +42,12 @@ func OrgRedirect() macaron.Handler {
 			return
 		}
 
-		newUrl := setting.ToAbsUrl(fmt.Sprintf("%s?%s", c.Req.URL.Path, c.Req.URL.Query().Encode()))
-		c.Redirect(newUrl, 302)
+		relURL := c.Req.URL.Path
+		if strings.HasPrefix(relURL, "/") {
+			relURL = relURL[1:]
```

Comment:
Thank you for taking the time to report and fix this issue.

I think this can be replaced with 
`relUrl = strings.TrimPrefix(relUlr, "/")`

Makes it more obvious what the code does. 
https://golang.org/pkg/strings/#TrimPrefix

Otherwise it looks ready to merge.